dashboard/modules/state/state_head.py

```diff
@@ -347,6 +371,40 @@ async def list_cluster_events(
         record_extra_usage_tag(TagKey.CORE_STATE_API_LIST_CLUSTER_EVENTS, "1")
         return await self._handle_list_api(self._state_api.list_cluster_events, req)
 
+    @routes.get("/api/v1/cluster_events")
+    @RateLimitedModule.enforce_max_concurrent_calls
+    async def list_events(self, req: aiohttp.web.Request) -> aiohttp.web.Response:
+        record_extra_usage_tag(TagKey.CORE_STATE_API_LIST_CLUSTER_EVENTS, "1")
+
+        job_id = req.query.get("job_id", None)
+        source_types = req.query.getall("sourceType", [])
+        severity_levels = req.query.getall("severityLevel", [])
+        count = int(req.query.get("count", 200))
+
+        # Filtering out specified keys from the query parameters and
+        # get {entity_name: entity_id}, for example, serve_app_name: "app"
+        excluded_keys = ["job_id", "sourceType", "severityLevel", "count"]
+        rest_of_query = {
+            key: value for key, value in req.query.items() if key not in excluded_keys
+        }
+
+        all_events = await self._state_api_data_source_client.get_all_events_as_list(
+            job_id
+        )
+
+        all_events = self.filter_events(
```

Comment:
Can you add an unit test for filter_events?

Reply:
Sounds good